server/qrops.go

```diff
@@ -28,177 +28,201 @@ func NewWriterCloserByteBuffer() *WriterCloserByteBuffer {
 	return &WriterCloserByteBuffer{bytes.NewBuffer([]byte{})}
 }
 
-func ToLogoImageFromBytes(imageBytes []byte, padding int) ([]byte, error) {
-	img, _, err := image.Decode(bytes.NewReader(imageBytes))
-	if err != nil {
-		return nil, fmt.Errorf("decoding image failed: %v", err)
-	}
-	paddedImg := images.AddPadding(img, padding)
-	circle := images.CreateCircle(img)
-	centeredImg := images.PlaceCircleInCenter(paddedImg, circle)
-	resultBytes, err := images.EncodePNG(centeredImg)
-	if err != nil {
-		return nil, fmt.Errorf("encoding PNG failed: %v", err)
-	}
-	return resultBytes, nil
+type QRConfig struct {
+	DecodedQRURL    string
+	WithLogo        bool
+	CorrectionLevel qrcode.EncodeOption
+	KeyUID          string
+	ImageName       string
+	Size            int
+	Params          url.Values
+	Logger          *zap.Logger
 }
 
-func GetLogoImage(multiaccountsDB *multiaccounts.Database, keyUID string, imageName string) ([]byte, error) {
-	var (
-		padding   int
-		LogoBytes []byte
-	)
-
-	staticImageData, err := images.Asset("_assets/tests/qr/status.png")
-	if err != nil { // Asset was not found.
-		return nil, err
-	}
-	identityImageObjectFromDB, err := multiaccountsDB.GetIdentityImage(keyUID, imageName)
-
-	if err != nil {
-		return nil, err
+func NewQRConfig(params url.Values, logger *zap.Logger) *QRConfig {
+	config := &QRConfig{}
+	config.Params = params
+	config.Logger = logger
+	config.SetQrURL()
+	config.SetAllowProfileImage()
+	config.SetErrorCorrectionLevel()
+	config.SetSize()
+
+	if config.WithLogo {
+		config.SetKeyUID()
+		config.SetImageName()
 	}
 
-	if identityImageObjectFromDB == nil {
-		padding = GetPadding(staticImageData)
-		LogoBytes, err = ToLogoImageFromBytes(staticImageData, padding)
-	} else {
-		padding = GetPadding(identityImageObjectFromDB.Payload)
-		LogoBytes, err = ToLogoImageFromBytes(identityImageObjectFromDB.Payload, padding)
-	}
-
-	return LogoBytes, err
-}
-
-func GetPadding(imgBytes []byte) int {
-	const (
-		defaultPadding = 20
-	)
-	size, _, err := images.GetImageDimensions(imgBytes)
-	if err != nil {
-		return defaultPadding
-	}
-	return size / 5
+	return config
 }
 
-func getQrURL(params url.Values, logger *zap.Logger) string {
-	qrURL, ok := params["url"]
+func (q *QRConfig) SetQrURL() {
+	qrURL, ok := q.Params["url"]
+	logger := q.Logger
```

Comment:
It isn't necessary to pass this pointer to a new variable.

You can just call `q.Logger.Error(...)`